### DISCUSSION TITLE
authentication - no Rpi desktop

### DISCUSSION THREAD

**Discussion by thatguyinoz** [skuethe/MMM-Spotify#133] [Q&A] [answered]
Hi
I have Magicmirror running in a very minimal environment with no desktop.
This means that i cannot complete the authentication part of the installation.
I have attempted to "fake" this by adding an additional redirect URI in the Spotify Dev account, then creating a URL manually and going to it on an other computer eg:
https://accounts.spotify.com/authorize?client_id=<XXXX>&response_type=code&redirect_uri=http://10.60.2.48:8888/callback
This created the username_token.json and populated it with some json data.
I added the basic config settings, and on restart, the mirror shows the spotify logo for a short time, but never populates song info when music is playing.
I do not seem to see error or debug info in the system logfiles (like pm2 logs) or in the debug console of the webbrowser.

Any clues that can be offered?

thanks

Peter

**Comment by skuethe** [comment, marked as answer]
Sorry about the late response, I was not watching new discussions on this repo :(
I just released a new version that fixes these auth problems in environments without a UI (docker f.e.).

Please have a look at the release `v2.1.0` and the updated `README` about docker setup / usage.

**Reply by LordRaccoon** [reply]
I am probably being very noob, but my auth also fails on docker and the guidance about the spotify.config.json was not enough for me to figure out what I did wrong, the message I get when running the auth is this:
 $ node first_auth.js
node:internal/modules/cjs/loader:1137
  throw err;
  ^

Error: Cannot find module 'express'
Require stack:
- /opt/mm/mounts/modules/MMM-Spotify/Spotify.js
- /opt/mm/mounts/modules/MMM-Spotify/first_auth.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
    at Module._load (node:internal/modules/cjs/loader:975:27)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/opt/mm/mounts/modules/MMM-Spotify/Spotify.js:5:17)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Module.require (node:internal/modules/cjs/loader:1225:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/mm/mounts/modules/MMM-Spotify/Spotify.js',
    '/opt/mm/mounts/modules/MMM-Spotify/first_auth.js'
  ]
}

**Reply by LordRaccoon** [reply]
oh, how stupid am I, I just installed the missing things and work now.

**Reply by LordRaccoon** [reply]
magicmirroos does not have a browser.. now I have a new problem

**Reply by skuethe** [reply]
There is documentation in the README how to handle it with a container setup (even with a docker compose example). Please have a look at that.
